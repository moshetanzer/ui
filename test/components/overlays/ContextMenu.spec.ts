import { describe, it, expect } from 'vitest'
import { UContextMenu } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('ContextMenu', () => {
  it.each([
    ['basic case', {}]
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof UContextMenu.props>) => {
    const html = await ComponentRender(nameOrHtml, options, UContextMenu)

    expect(html).toMatchSnapshot()
  })
})
