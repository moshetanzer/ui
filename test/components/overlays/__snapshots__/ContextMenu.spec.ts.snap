// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContextMenu > renders basic case correctly 1`] = `
"<div class="z-20 group">
  <transition-stub enteractiveclass="transition ease-out duration-200" enterfromclass="opacity-0 translate-y-1" entertoclass="opacity-100 translate-y-0" leaveactiveclass="transition ease-in duration-150" leavefromclass="opacity-100 translate-y-0" leavetoclass="opacity-0 translate-y-1" appear="true" persisted="false" css="true">
    <div>
      <!--v-if-->
      <div class="overflow-hidden focus:outline-none relative ring-1 ring-gray-200 dark:ring-gray-800 rounded-md shadow-lg bg-white dark:bg-gray-900"></div>
    </div>
  </transition-stub>
</div>"
`;
